definitions/checks/unread_mail.rb

```diff
@@ -0,0 +1,46 @@
+class Checks::UnreadMail < ForemanMaintain::Check
+  metadata do
+    label :unread_mail
+    description "Check for mails in root's mailbox."
+    tags :unread_mail
+  end
+
+  attr_reader :file_path, :mailbox_dir
+
+  def initialize
+    @mailbox_dir = find_mailbox_directory
+    @file_path = root_mail_file_path
+  end
+
+  def run
+    if file_exists?(file_path)
+      mcount = mails_count
+      assert(mcount == 0, "WARNING: Found #{mcount} mail(s) in root's mailbox.",
+             :next_steps => [Procedures::SystemMailbox::List.new(:file_path => file_path),
+                             Procedures::SystemMailbox::Clear.new(:file_path => file_path)],
+             :error_type => ForemanMaintain::Error::Warn)
+    end
+  end
+
+  private
+
```

Comment:
@kgaikwad, would it make sense to make `file_path` and `mailbox_dir` lazy so that they are evaluated during the `run` instead of initialization? I see two advantages there:
 - it takes time during scenario execution when there is already something printed out and scenario can start quicker
 - if (hypothetically) if file_path is subject of change in some of the previous steps we won't notice

Reply:
Yes, make sense! And advantages mentioned are legit.
I will do modifications as suggested.